netconf_client/ncclient.py

```diff
@@ -21,40 +25,201 @@
 )
 
 
+# Defines the scope for netconf traces
+_logger = logging.getLogger("netconf_client.manager")
+
+
+def _pretty_xml(xml):
+    """Reformats a given string containing an XML document (for human readable output)"""
+
+    parser = etree.XMLParser(remove_blank_text=True)
+    tree = etree.fromstring(xml, parser)
+    return etree.tostring(tree, pretty_print=True).decode()
+
+
 class Manager:
-    """A helper class for performing common NETCONF operations
+    """A helper class for performing common NETCONF operations with pretty logging.
 
     This class attempts to be API compatible with the manager object
     from the original ncclient.
 
     This class is also a context manager and can be used with `with`
     statements to automatically close the underlying session.
 
+    NetConf requests and responses are logged using the ``netconf_client.manager`` scope.
```

Comment:
Sorry this is a common nitpick of mine
```suggestion
    NETCONF requests and responses are logged using the ``netconf_client.manager`` scope.
```